subprojects/core/src/main/java/org/gradle/api/tasks/JavaExec.java

```diff
@@ -279,6 +282,18 @@ public JavaExec setMain(String mainClassName) {
         return javaExecHandleBuilder.getArgs();
     }
 
+    /**
+     * Command line arguments passed to the main class.
+     * @param args the command line arguments as a single string. Will be parsed to argument list.
+     * @return this
+     * @since 4.9
+     */
+    @Incubating
+    @Option(option = "args", description = "Command line arguments passed to the main class. [INCUBATING]")
+    public JavaExec setArgs(String args) {
+        return setArgs(Arrays.asList(Commandline.translateCommandline(args)));
```

Comment:
I think this allows someone to do...
```
task run(type: JavaExec) {
   args = "these are my arguments"
}
```
Where we would split "these are my arguments".  We would rather someone do `args = [ "these", "are", "my", "arguments" ]`.  Maybe this should be called something else (e.g., `setCommandLineUserArguments`) with a note about how this is not expected to be called directly?

Reply:
And should this overwrite the arguments from the build script or append to them?  Either way could make sense and we should have coverage for whichever way we choose.

Reply:
Thanks @big-guy, a really nice catch! It's renamed to `setArgsString`. And I think one of current tests already clarify it: `arguments passed via command line take precedence`